core/src/main/java/com/linecorp/armeria/client/Http1ResponseDecoder.java

```diff
@@ -194,10 +202,32 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
                         assert res != null;
                         this.res = res;
 
-                        res.startResponse();
                         final ResponseHeaders responseHeaders = ArmeriaHttpUtil.toArmeria(nettyRes);
+                        final HttpStatus status = responseHeaders.status();
+
+                        if (res.needs100Continue()) {
+                            if (status == HttpStatus.CONTINUE) {
+                                res.resume();
+                            } else if (status == HttpStatus.EXPECTATION_FAILED) {
+                                // Remove the response and do not close the HttpResponseWrapper in order not to
+                                // close the original response.
+                                removeResponse(resId++);
+                                if (res.repeat()) {
+                                    state = State.DISCARD_DATA_OR_TRAILERS;
+                                    // TODO(minwoox): reset timeout
+                                } else {
+                                    // session is not acquirable.
+                                    state = State.DISCARD;
+                                }
+                                return;
+                            } else {
+                                res.discardRequestBody();
+                            }
+                        }
+                        res.startResponse();
```

Comment:
Question) Could we move this logic to `HttpResponseWrapper`? It is unlikely that 100-continue will be implemented differently for a session protocol. It would be nice if we could delete the duplicate code in `Http{1,2}ResponseDecoder` and handle one place.

Reply:
I had no choice because I had to call `removeResponse` when `EXPECTATION_FAILED` is received.
However, we have decided not to implement the repeat so I can place the logic in `AbstractHttpRequestHander` now.